src/Core/Grid/DataProvider/LogGridDataProvider.php

```diff
@@ -0,0 +1,112 @@
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\DataProvider;
+
+use Doctrine\DBAL\Driver\Connection;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use PrestaShopBundle\Entity\Repository\LogRepository;
+use PrestaShopBundle\Service\Hook\HookDispatcher;
+
+/**
+ * Class LogsGridDataProvider is responsible for providing data for Logs grid
+ */
+final class LogGridDataProvider implements GridDataProviderInterface
+{
+    /**
+     * @var LogRepository
+     */
+    private $logRepository;
+
+    /**
+     * @var HookDispatcher
+     */
+    private $hookDispatcher;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @param LogRepository $logRepository
+     * @param HookDispatcher $hookDispatcher
+     * @param Connection $connection
+     */
+    public function __construct(
+        LogRepository $logRepository,
+        HookDispatcher $hookDispatcher,
+        Connection $connection
+    ) {
+        $this->logRepository = $logRepository;
+        $this->hookDispatcher = $hookDispatcher;
+        $this->connection = $connection;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRows(SearchCriteriaInterface  $searchCriteria)
+    {
+        $logSqlQuery = $this->getQuery($searchCriteria);
+
+        $stmt = $this->connection->prepare($logSqlQuery);
+        $stmt->execute();
+
+        $logs = $stmt->fetchAll(\PDO::FETCH_ASSOC);
+
+        return $logs;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRowsTotal()
+    {
+        return count($this->logRepository->findAll());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getQuery(SearchCriteriaInterface $searchCriteria)
+    {
+        $logQuery = $this->logRepository->getAllWithEmployeeInformationQuery([
+            'offset' => $searchCriteria->getOffset(),
+            'limit' => $searchCriteria->getLimit(),
+            'filters' => $searchCriteria->getFilters(),
+            'orderBy' => $searchCriteria->getOrderBy(),
+            'sortOrder' => $searchCriteria->getOrderWay(),
+        ]);
+
+        $this->hookDispatcher->dispatchForParameters('modifyLogGridQuery', [
+            'query' => $logQuery,
```

Comment:
this is `query_builder` and `modifyLogGridQueryBuilder`, note this class is a good candidate to be refactored later :)